src/Common/src/System/Collections/Generic/ReferenceEqualityComparer.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Collections.Generic
+{
+    internal sealed class ReferenceEqualityComparer<T> : IEqualityComparer<T>
+    {
+        internal static readonly ReferenceEqualityComparer<T> Instance = new ReferenceEqualityComparer<T>();
```

Comment:
Nit: we actually already have one of these in the repo:
https://github.com/dotnet/corefx/blob/master/src/System.Linq.Expressions/src/System/Dynamic/Utils/ReferenceEqualityComparer.cs
Could we consolidate?